app/controllers/users/phone_setup_controller.rb

```diff
@@ -73,30 +73,27 @@ def setup_voice_preference?
 
     def handle_create_success(phone)
       if MfaContext.new(current_user).phone_configurations.map(&:phone).index(phone).nil?
-        prompt_to_confirm_phone(
-          id: nil,
-          phone: @new_phone_form.phone,
-          selected_delivery_method: @new_phone_form.otp_delivery_preference,
-          phone_type: @new_phone_form.phone_info&.type,
-          selected_default_number: @new_phone_form.otp_make_default_number,
-        )
+        prompt_to_confirm_phone
       else
         flash[:error] = t('errors.messages.phone_duplicate')
         redirect_to phone_setup_url
       end
     end
 
-    def prompt_to_confirm_phone(id:, phone:, selected_delivery_method: nil,
-                                selected_default_number: nil, phone_type: nil)
-
-      user_session[:unconfirmed_phone] = phone
+    def prompt_to_confirm_phone
+      user_session[:unconfirmed_phone] = @new_phone_form.phone
       user_session[:context] = 'confirmation'
-      user_session[:phone_type] = phone_type.to_s
+      user_session[:phone_type] = @new_phone_form.phone_info&.type.to_s
+      user_session[:phone_recaptcha_assessment_id] = @new_phone_form.recaptcha_assessment_id
 
       redirect_to otp_send_url(
         otp_delivery_selection_form: {
-          otp_delivery_preference: otp_delivery_method(id, phone, selected_delivery_method),
-          otp_make_default_number: selected_default_number,
+          otp_delivery_preference: otp_delivery_method(
+            nil,
+            @new_phone_form.phone,
+            @new_phone_form.otp_delivery_preference,
+          ),
```

Comment:
maybe it's time to refactor `otp_delivery_method`? It's only used in this file, and the first argument is completely dropped anyways? Seems like a good use case for keyword args too IMO

Could also be a good follow-up option

Reply:
Yeah I did a light refactoring here since this method `prompt_to_confirm_phone` was strangely parameterized despite being unnecessary to do so, making it difficult to extend. Maybe this method and `otp_delivery_method` were reused previously, when there were separate controllers between account creation vs. dashboard phone creation.

There's a lot of this phone stuff I'd like to refactor (especially phone-specific stuff in `TwoFactorAuthenticationController`), so probably save for a follow-up.